STM32G0_Nucleo/Core/Src/main.c

```diff
@@ -0,0 +1,296 @@
+/* USER CODE BEGIN Header */
+/**
+  ******************************************************************************
+  * @file           : main.c
+  * @brief          : Main program body
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2024 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+#include <stdio.h>
+#include <string.h>
+
+#ifndef WOLFSSL_USER_SETTINGS
+    #include <wolfssl/options.h>
+#endif
+
+#include <wolfssl/wolfcrypt/pwdbased.h>  // For wc_PBKDF2_HMAC
+#include <wolfssl/wolfcrypt/aes.h>       // For wc_AesSetKeyDirect
+#include <user_settings.h>

```

Comment:
Please move includes for settings.h and options.h above any other wolfSSL headers.
options.h is only generated if building the wolfSSL from ./configure or cmake. Your use-cases should be covered with including wolfssl/wolfcrypt/settings.h. 
If you are not using the Cube MX HAL wolfSSL pack then make sure your project defines WOLFSSL_USER_SETTINGS.
See https://github.com/wolfSSL/wolfssl/blob/master/wolfssl/wolfcrypt/settings.h#L332